test/libyul/YulOptimizerTest.cpp

```diff
@@ -235,6 +236,17 @@ TestCase::TestResult YulOptimizerTest::run(ostream& _stream, string const& _line
 		ExpressionJoiner::run(*m_context, *m_ast);
 		ExpressionJoiner::run(*m_context, *m_ast);
 	}
+	else if (m_optimizerStep == "unusedFunctionParameterPruner")
+	{
+		disambiguate();
+		// prerequisite for SSATransform
+		ForLoopInitRewriter::run(*m_context, *m_ast);
+		SSATransform::run(*m_context, *m_ast);
+		FunctionHoister::run(*m_context, *m_ast);
+		ForLoopConditionIntoBody::run(*m_context, *m_ast);
+		ExpressionSplitter::run(*m_context, *m_ast);
+		UnusedFunctionParameterPruner::run(*m_context, *m_ast);
```

Comment:
I think it's a good idea to reverse `ExpressionSplitter` and `SSATransform` at the end, so tests are more readable.